src/Microsoft.Health.ControlPlane.Core/Features/Rbac/RbacService.cs

```diff
@@ -23,16 +25,35 @@ public RbacService(IControlPlaneDataStore controlPlaneDataStore)
 
         public async Task<IdentityProvider> GetIdentityProviderAsync(string name, CancellationToken cancellationToken)
         {
-            EnsureArg.IsNotNullOrWhiteSpace(name, nameof(name));
-
             return await _controlPlaneDataStore.GetIdentityProviderAsync(name, cancellationToken);
         }
 
         public async Task<IdentityProvider> UpsertIdentityProviderAsync(IdentityProvider identityProvider, CancellationToken cancellationToken)
         {
-            EnsureArg.IsNotNull(identityProvider, nameof(identityProvider));
-
             return await _controlPlaneDataStore.UpsertIdentityProviderAsync(identityProvider, cancellationToken);
         }
+
+        public async Task<Role> GetRoleAsync(string name, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNullOrWhiteSpace(name, nameof(name));
+            return await _controlPlaneDataStore.GetRoleAsync(name, cancellationToken);
+        }
+
+        public async Task<IEnumerable<Role>> GetRoleForAllAsync(CancellationToken cancellationToken)
+        {
+            return await _controlPlaneDataStore.GetRoleAllAsync(cancellationToken);
+        }
+
+        public async Task<Role> UpsertRoleAsync(Role role, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(role, nameof(role));
+            return await _controlPlaneDataStore.UpsertRoleAsync(role, cancellationToken);
```

Comment:
We should validate the role here. There's already a `Validate()` method on the existing `Role.cs` in `Fhir.Core.Features.Security.Role` that we can use/refactor. 

While we're at it, we should probably remove the existing one and refactor to use this. Same goes with the `ResourcePermission.cs` etc.

Reply:
Fixed this one